Error installing

Looks like the download is no longer valid.

```
> phantomjs@1.9.7-6 install /Users/troywhiteley/Projects/smooth-scroller/node_modules/phantomjs
> node install.js

Downloading http://cdn.bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.7-macosx.zip
Saving to /var/folders/fl/ssqg11_j1hs5mk4fvyqgjcmc0000gn/T/phantomjs/phantomjs-1.9.7-macosx.zip
Receiving...
Error requesting archive.
Status: 403
Request options: {
  "protocol": "http:",
  "slashes": true,
  "auth": null,
  "host": "cdn.bitbucket.org",
  "port": null,
  "hostname": "cdn.bitbucket.org",
  "hash": null,
  "search": null,
  "query": null,
  "pathname": "/ariya/phantomjs/downloads/phantomjs-1.9.7-macosx.zip",
  "path": "/ariya/phantomjs/downloads/phantomjs-1.9.7-macosx.zip",
  "href": "http://cdn.bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.7-macosx.zip"
}
Response headers: {
  "content-type": "application/xml",
  "transfer-encoding": "chunked",
  "connection": "keep-alive",
  "date": "Fri, 23 May 2014 05:52:36 GMT",
  "server": "AmazonS3",
  "age": "105",
  "x-cache": "Error from cloudfront",
  "via": "1.1 a6f2756cb20b30d5bd71ce0dbeb6e8fe.cloudfront.net (CloudFront)",
  "x-amz-cf-id": "AMfg2Dv3gvWiHrG2Nc_kt_xZFEqkRyAij0IR1u_JPJncS7wDbj5OAw=="
}
Make sure your network and proxy settings are correct.
npm ERR! phantomjs@1.9.7-6 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the phantomjs@1.9.7-6 install script.
npm ERR! This is most likely a problem with the phantomjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls phantomjs
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.2.0
npm ERR! command "node" "/usr/local/bin/npm" "install" "phantomjs"
npm ERR! cwd /Users/troywhiteley/Projects/smooth-scroller
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/troywhiteley/Projects/smooth-scroller/npm-debug.log
npm ERR! not ok code 0
```

--- a/install.js
+++ b/install.js
@@ -10,18 +10,18 @@ var AdmZip = require('adm-zip')
 var cp = require('child_process')
 var fs = require('fs')
 var helper = require('./lib/phantomjs')
-var http = require('http')
 var kew = require('kew')
+var mkdirp = require('mkdirp')
 var ncp = require('ncp')
 var npmconf = require('npmconf')
-var mkdirp = require('mkdirp')
 var path = require('path')
+var request = require('request')
 var rimraf = require('rimraf').sync
 var url = require('url')
 var util = require('util')
 var which = require('which')
 
-var cdnUrl = process.env.PHANTOMJS_CDNURL || 'http://cdn.bitbucket.org/ariya/phantomjs/downloads'
+var cdnUrl = process.env.PHANTOMJS_CDNURL || 'https://bitbucket.org/ariya/phantomjs/downloads'
 var downloadUrl = cdnUrl + '/phantomjs-' + helper.version + '-'
 
 var originalPath = process.env.PATH
@@ -116,10 +116,10 @@ whichDeferred.promise
         path.join(pkgPath, 'bin' ,'phantomjs')
     var relativeLocation = path.relative(libPath, location)
     writeLocationFile(relativeLocation)
-    
+
     // Ensure executable is executable by all users
     fs.chmodSync(location, '755')
-    
+
     console.log('Done. Phantomjs binary available at', location)
     exit(0)
   })
@@ -178,25 +178,30 @@ function findSuitableTempDirectory(npmConf) {
 
 
 function getRequestOptions(conf) {
-  var proxyUrl = conf.get('http-proxy') || conf.get('proxy')
+  var options = {
+    uri: downloadUrl,
+    encoding: null, // Get response as a buffer
+    followRedirect: true // The default download path redirects to a CDN URL.
+  }
 
+  var proxyUrl = conf.get('http-proxy') || conf.get('proxy')
   if (proxyUrl) {
     console.log('Using proxy ' + proxyUrl)
-    var options = url.parse(proxyUrl)
-    options.path = downloadUrl
-    options.headers = { Host: url.parse(downloadUrl).host }
-    // If going through proxy, spoof the User-Agent, since may commerical proxies block blank or unknown agents in headers
-    options.headers['User-Agent'] = 'curl/7.21.4 (universal-apple-darwin11.0) libcurl/7.21.4 OpenSSL/0.9.8r zlib/1.2.5'
+    var proxy = url.parse(proxyUrl)
+
     // Turn basic authorization into proxy-authorization.
-    if (options.auth) {
-      options.headers['Proxy-Authorization'] = 'Basic ' + new Buffer(options.auth).toString('base64')
-      delete options.auth
+    if (proxy.auth) {
+      options.headers['Proxy-Authorization'] = 'Basic ' + new Buffer(proxy.auth).toString('base64')
+      delete proxy.auth
     }
 
-    return options
-  } else {
-    return url.parse(downloadUrl)
+    options.proxy = url.format(proxy)
+
+    // If going through proxy, spoof the User-Agent, since may commerical proxies block blank or unknown agents in headers
+    options.headers['User-Agent'] = 'curl/7.21.4 (universal-apple-darwin11.0) libcurl/7.21.4 OpenSSL/0.9.8r zlib/1.2.5'
   }
+
+  return options
 }
 
 
@@ -208,34 +213,30 @@ function requestBinary(requestOptions, filePath) {
   var writePath = filePath + '-download-' + Date.now()
   var outFile = fs.openSync(writePath, 'w')
 
-  var client = http.get(requestOptions, function (response) {
-    var status = response.statusCode
-    console.log('Receiving...')
+  console.log('Receiving...')
+  request(requestOptions, function (error, response, body) {
+    if (!error && response.statusCode === 200) {
+      fs.writeFileSync(writePath, body)
+      console.log('Received ' + Math.floor(body.length / 1024) + 'K total.')
+      fs.renameSync(writePath, filePath)
+      deferred.resolve(filePath)
 
-    if (status === 200) {
-      response.addListener('data',   function (data) {
-        fs.writeSync(outFile, data, 0, data.length, null)
-        count += data.length
-        if ((count - notifiedCount) > 800000) {
-          console.log('Received ' + Math.floor(count / 1024) + 'K...')
-          notifiedCount = count
-        }
-      })
-
-      response.addListener('end',   function () {
-        console.log('Received ' + Math.floor(count / 1024) + 'K total.')
-        fs.closeSync(outFile)
-        fs.renameSync(writePath, filePath)
-        deferred.resolve(filePath)
-      })
-
-    } else {
-      client.abort()
+    } else if (response) {
       console.error('Error requesting archive.\n' +
-          'Status: ' + status + '\n' +
+          'Status: ' + response.statusCode + '\n' +
           'Request options: ' + JSON.stringify(requestOptions, null, 2) + '\n' +
           'Response headers: ' + JSON.stringify(response.headers, null, 2) + '\n' +
-          'Make sure your network and proxy settings are correct.')
+          'Make sure your network and proxy settings are correct.\n\n' +
+          'If you continue to have issues, please report this full log at ' +
+          'https://github.com/Medium/phantomjs')
+      exit(1)
+    } else if (error) {
+      console.error('Error making request.\n' + error.stack + '\n\n' +
+          'Please report this full log at https://github.com/Medium/phantomjs')
+      exit(1)
+    } else {
+      console.error('Something unexpected happened, please report this full ' +
+          'log at https://github.com/Medium/phantomjs')
       exit(1)
     }
   })
@@ -264,7 +265,7 @@ function extractDownload(filePath) {
       zip.extractAllTo(extractedPath, true)
       deferred.resolve(extractedPath)
     } catch (err) {
-      console.error('Error extracting archive')
+      console.error('Error extracting zip')
       deferred.reject(err)
     }
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "phantomjs",
-  "version": "1.9.7-6",
+  "version": "1.9.7-7",
   "keywords": [
     "phantomjs",
     "headless",
@@ -44,6 +44,7 @@
     "ncp": "0.4.2",
     "npmconf": "0.0.24",
     "mkdirp": "0.3.5",
+    "request": "2.36.0",
     "rimraf": "~2.2.2",
     "which": "~1.0.5"
   },

